### DISCUSSION TITLE
Lack of Query Builder and custom fields

### DISCUSSION THREAD

**Discussion by itspers** [prisma/prisma#17307] [Q&A]
Few month ago while starting new project i checked 'whats new on the market' and pick Prisma as new DB tool instead of traditional TypeORM for NestJS. Checked website, thought 'nice, little less code boilerplate, simpler migrations' and didnt even bother to check whole website docs - thought something like Query Builder will be on the bottom. 

And now, after creating dozens of pages using methods for creating and editing entities found out that i can only query lists, with filters and sort. But in reality application for each add entity method will have 10 grids and charts with custom joins and fields. 

And all similar issues leads to 'this will be not type safe - use raw query'. 

So is it because project originally came from mongo+graphql with 'here is tree of data, calculate whatever you need on client'? 

Maybe people solving this somehow with using something like knex on top of Prisma? With so much stars on github, it cannot be that people just writing raw SQL in real-world apps or making multiple queries and join data in JS...

For example: i have Users and table that holds WeeklyWorkingDays - userId, dateFrom and value (Mon,Tue,Wed..). Creating, saving this records is nice. But now i need to have current working days for user in grid. But i cannot just run findMany with custom join and select field as... And there will be dozens fields like that in multiple grids and charts...

After searching throw this discussion and issues i had idea that better to rewrite everything to TypeORM while its 10 tables and not 100, but maybe somebody know how to deal with it right?

**Comment by nurul3101** [comment]
Hello @itspers 👋 

You can use Knex along with Prisma for queries that are not natively supported by Prisma yet. I have seen several users use a combination of Knex and Prisma. We have a feature request for mentioning the [Join attribute](https://github.com/prisma/prisma/issues/5184), would something like this solve the issue for you?

I believe you are referring to this [GitHub Issue](https://github.com/prisma/prisma/issues/4997) when referring to making multiple queries and joining data?

**Reply by itspers** [reply]
Thanks @nurul3101 , 

Unfortunately making join option will solve only lack of join, but when you join - is likely you will want to select something from that join... or group, or subselect...  Ideal solution would be proper query builder. So documentation will say "ok, you can simple insert/update/findMany with filters, for everything else use 'prisma.queryBuilder.from('tableA').join('tableB').select('a','b','tableB.c').where(...'"

Knex will have no autocomplete and typings, so i for now have plan to take https://github.com/koskimas/kysely, feed it somehow with prisma types and make run rawQuery using prisma. In theory this will solve my current problems.

But im still have concern in general about using Prisma. On one hand simple insert/update/delete operations are good enough and have transactions, migration process also nice, in combination with good query builder - should be good solution with less code. Plus developers working for money will do much more in future in comparison to other libs. On the other hand target audience of future futures will be startups writing next.js hello-world apps, so more likely there will be more findMany options for Graphql, but still lack of down migrations and no js-migration files...

So im still have this fear that i would write tone of code, 100 tables and then will need to rewrite all that to good old TypeORM when will need to do something serious

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
